### ISSUE TITLE
Enhancement: Default allowAutomaticSingleRunInference to true in v6

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6176]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

Since #3512 -> #3528 added the opt-in `allowAutomaticSingleRunInference`, there have been relatively few changes to the flag. AFAIK the only bugs reported have been:

* #3626: fixed by #3655
* #4435: "resolved" by indicating the user needs to be more granular in specifying projects
* #5880: fixed by #5893

_Shoutout @aaronadamsCA for reporting 2/3 of those!_ 😄 

It seems to me like the flag is pretty stable. Since we'll be asking the public to test out the upcoming v6 major version Soon™️, that new version seems like a good time to make `allowAutomaticSingleRunInference` opt-in as a breaking change. 

### Fail

```typescript
n/a
```

### Pass

```typescript
n/a
```

**Comment by bradzacher** [comment]
It's a bit funky with `--fix` from what I've noticed.
The fix re-runs won't have proper type information so rules like the `no-unsafe-*` rules will report false errors on fixed files that won't show up on a non-fixing run.

We need to figure out a better solution for the fix case before we can really roll this out to everyone I think, or else we'll create a really flaky and confusing experience.

**Comment by JoshuaKGoldberg** [comment]
Got it - I'll close this and file a separate bug. Thanks!
